js/list.js

```diff
@@ -1,452 +1,8 @@
-        description: 'In the antebellum United States, Solomon Northup, a free black man from upstate New York, is abducted and sold into slavery.',
-        movie_year: 2013,
-        director: "Steve McQueen",
-        actors: ["Chiwetel Ejiofor", "Michael Kenneth Williams", "Michael Fassbender", "Brad Pitt", "Benedict Cumberbatch"],
-        poster_url: 'https://www.themoviedb.org/t/p/w600_and_h900_bestv2/xdANQijuNrJaw1HA61rDccME4Tm.jpg',
-        price: '90.00'
-    },
-    {
-        id: 42,
-        title: 'The Shining',
-        description: 'A family heads to an isolated hotel for the winter where a sinister presence influences the father into violence, while his psychic son sees horrific forebodings from both past and future.',
-        movie_year: 1980,
-        director: "Stanley Kubrick",
-        actors: ["Jack Nicholson", "Shelley Duvall", "Danny Lloyd", "Scatman Crothers", "Barry Nelson"],
-        poster_url: 'https://www.themoviedb.org/t/p/w1280/upBsuweJjBftrkJGqVf8PHCdH0l.jpg',
-        price: '85.00'
-    },
-    {
-        id: 43,
-        title: 'Fight Club',
-        description: 'An insomniac office worker and a devil-may-care soap maker form an underground fight club that evolves into much more.',
-        movie_year: 1999,
-        director: "David Fincher",
-        actors: ["Brad Pitt", "Edward Norton", "Meat Loaf", "Zach Grenier", "Holt McCallany"],
-        poster_url: 'https://www.themoviedb.org/t/p/w600_and_h900_bestv2/pB8BM7pdSp6B6Ih7QZ4DrQ3PmJK.jpg',
-        price: '85.00'
-    },
-    {
-        id: 44,
-        title: 'Schindler\'s List',
-        description: 'In German-occupied Poland during World War II, industrialist Oskar Schindler gradually becomes concerned for his Jewish workforce after witnessing their persecution by the Nazis.',
-        movie_year: 1993,
-        director: "Steven Spielberg",
-        actors: ["Liam Neeson", "Ralph Fiennes", "Ben Kingsley", "Caroline Goodall", "Jonathan Sagall"],
-        poster_url: 'https://www.themoviedb.org/t/p/w600_and_h900_bestv2/sF1U4EUQS8YHUYjNl3pMGNIQyr0.jpg',
-        price: '95.00'
-    },
-    {
-        id: 45,
-        title: 'Forrest Gump',
-        description: 'The presidencies of Kennedy and Johnson, the events of Vietnam, Watergate, and other historical events unfold from the perspective of an Alabama man with an IQ of 75, whose only desire is to be reunited with his childhood sweetheart.',
-        movie_year: 1994,
-        director: "Robert Zemeckis",
-        actors: ["Tom Hanks", "Robin Wright", "Gary Sinise", "Sally Field", "Mykelti Williamson"],
-        poster_url: 'https://www.themoviedb.org/t/p/w600_and_h900_bestv2/saHP97rTPS5eLmrLQEcANmKrsFl.jpg',
-        price: '85.00'
-    }
-];
\ No newline at end of file
+function fetchAllMovies(callback) {
+    fetch("https://raw.githubusercontent.com/halynabondar/halynabondar.github.io/main/Data/movieApp.json")
+        .then(response => {
+            response.json().then(moviesList => {
```

Comment:
You have the beginning of a [callback hell](https://medium.com/@raihan_tazdid/callback-hell-in-javascript-all-you-need-to-know-296f7f5d3c1) here. You can make your code more readable by chaining `.then` callbacks on the same level.
Also, it's a good practice to add a `catch` to your Promise so you can handle errors nicely.

```js
fetch("...")
  .then((response) => {
    return response.json();
  })
  .then((moviesList) => {
    callback(moviesList);
  })
  .catch((error) => {
    console.error(error)
  });
```

Reply:
Fixed